### PR TITLE
e2e/autoscaler: consume MaxNodesTotalReached event

### PR DESCRIPTION
**Pull request by frobware**
The previous behaviour was to scale out by waiting for the expected
scale up events to be generated on the node groups. After that had
completed we then waited for a further 60s to ensure that there were
no further scale up attempts as we had explicitly capped the cluster
size.

This commit changes the behaviour from waiting for 60s to explicitly
waiting for the MaxNodesTotalReached event to be generated as well. We use
that event to assert that the cluster will not grow beyond the
specified maximum.

This PR also helps to validate that we carry two upstream fixes that
were made with respect to calculating when the maximum size of the
cluster is reached:
    - https://github.com/kubernetes/autoscaler/pull/1516
    - https://github.com/kubernetes/autoscaler/pull/1771

### CODE REVIEW

File: pkg/e2e/autoscaler/autoscaler.go [hunk 1]
```diff
@@ -282,21 +295,15 @@ var _ = g.Describe("[Feature:Machines][Serial] Autoscaler should", func() {
...
-		// capped size.
-		//
-		// TODO(frobware): switch to matching on
-		// MaxNodesTotalReached when that is available in the
-		// cluster-autoscaler image.
+		// cluster size with maxNodesTotal. We check that we
+		// received the max cluster size reached event.
+		maxNodesTotalReachedCounter := newMaxNodesTotalReachedCounter(eventWatcher, 0)
 		testDuration = time.Now().Add(time.Duration(e2e.WaitShort))
-		o.Consistently(func() bool {
```
Comment (enxebre) [inline pkg/e2e/autoscaler/autoscaler.go]: isn't this complementary check?

Comment (frobware) [reply pkg/e2e/autoscaler/autoscaler.go]: > isn't this complementary check?

Yes, it's back in.

**Comment by frobware** [comment]
/hold cancel

This merged: https://github.com/openshift/kubernetes-autoscaler/pull/102

**Comment by enxebre** [comment]
test in this repo were running against out of date release payload hence not picking latest autoscaler changes.
This should be fixed now https://github.com/openshift/release/pull/4000

**Comment by frobware** [comment]
/hold

don't want this to merge just yet.

**Comment by frobware** [comment]
/hold 

Until I work out why this is failing on k8s-e2e.

**Comment by frobware** [comment]
/hold

Waiting for https://github.com/openshift/kubernetes-autoscaler/pull/103

**Pull request merged**